dlt/destinations/impl/lancedb/configuration.py

```diff
@@ -0,0 +1,102 @@
+import dataclasses
+from typing import Optional, Final, Literal, ClassVar, List
+
+from dlt.common.configuration import configspec
+from dlt.common.configuration.specs.base_configuration import (
+    BaseConfiguration,
+    CredentialsConfiguration,
+)
+from dlt.common.destination.reference import DestinationClientDwhConfiguration
+from dlt.common.typing import TSecretStrValue
+from dlt.common.utils import digest128
+
+
+@configspec
+class LanceDBCredentials(CredentialsConfiguration):
+    uri: str = "./.lancedb"
+    """LanceDB database URI. Defaults to local, on-disk instance.
+
+    The available schemas are:
+
+    - `/path/to/database` - local database.
+    - `db://host:port` - remote database (LanceDB cloud).
+    """
+    api_key: TSecretStrValue = None
+    """API key for the remote connections (LanceDB cloud)."""
+    embedding_model_provider_api_key: str = None
+    """API key for the embedding model provider."""
+
+    __config_gen_annotations__: ClassVar[List[str]] = [
+        "uri",
+        "api_key",
+        "embedding_model_provider_api_key",
+    ]
+
+
+@configspec
+class LanceDBClientOptions(BaseConfiguration):
+    max_retries: Optional[int] = 1
+    """`EmbeddingFunction` class wraps the calls for source and query embedding
+    generation inside a rate limit handler that retries the requests with exponential
+    backoff after successive failures.
+
+    You can tune it by setting it to a different number, or disable it by setting it to 0."""
+
+    __config_gen_annotations__: ClassVar[List[str]] = [
+        "max_retries",
+    ]
+
+
+TEmbeddingProvider = Literal[
+    "gemini-text",
+    "bedrock-text",
+    "cohere",
+    "gte-text",
+    "imagebind",
+    "instructor",
+    "open-clip",
+    "openai",
+    "sentence-transformers",
+    "huggingface",
+    "colbert",
+]
+
+
+@configspec
+class LanceDBClientConfiguration(DestinationClientDwhConfiguration):
+    destination_type: Final[str] = dataclasses.field(  # type: ignore
+        default="LanceDB", init=False, repr=False, compare=False
+    )
+    credentials: LanceDBCredentials = None
+    dataset_separator: str = "_"
```

Comment:
imho we should have a triple underscore here. if datasets have an underscore in their name, we might start dropping tables that are in the wrong "dataset" this way.

Reply:
Fixed!